dnd-troops-api/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/api/dto/UserDTO.java

```diff
@@ -0,0 +1,99 @@
+package cz.muni.fi.pa165.w2018.dndtroops.api.dto;
+
+import cz.muni.fi.pa165.w2018.dndtroops.api.enums.UserType;
+
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+/**
+ * User Data Transfer Object
+ *
+ * @author Marek Valko
+ */
+public class UserDTO {
+
+    @NotNull
+    private Long id;
+
+    @NotNull
+    private String login;
+
+    @NotNull
+    private UserType type;
+
+    @NotNull
+    private String passwordHash;
+
+    private HeroDTO heroDTO;
+
+    public UserDTO() {
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Move equals and hashCode before toString after getters and setters for consistency reasons